e2e/cypress/integration/team/teams_spec.js

```diff
@@ -132,15 +165,13 @@ describe('Teams Suite', () => {
         cy.loginAsNewUser();
 
         // # Leave all teams
-        function leaveAllTeams() {
-            cy.apiGetTeams().then((response) => {
-                if (response.body.length > 0) {
-                    cy.leaveTeam().then(() => leaveAllTeams());
-                }
+        cy.apiGetTeams().then((response) => {
+            response.body.forEach((team) => {
+                cy.visit(`/${team.name}/channels/town-square`);
+                cy.get('#headerTeamName').should('be.visible').and('have.text', team.display_name);
+                cy.leaveTeam();
```

Comment:
Instead of undeterministic removal of team, specifically delete each team where the user is member of.